### PR TITLE
Ask people Api call for setting up new survery 

### PR DESCRIPTION
**Pull request by markphi2013**
run aco_sync for the new controller
Json post call to add new program:
```json
{
  "survey": {
    "uid": "survey_uid0001",
    "participants_url": "http://askpeople.com/surveys/participants/survey_uid0001/list.csv",
    "questions": [
      {
        "uid": "question_uid001",
        "question_text": "example question 1?",
        "answers": [
          {
            "uid": "answer_uid_001",
            "answer_text": "yes"
          },
          {
            "uid": "answer_uid_002",
            "answer_text": "no"
          }
        ]
      },
      {
        "uid": "question_uid002",
        "question_text": "example question 2?",
        "answers": [
          {
            "uid": "answer_uid_003",
            "answer_text": "yes"
          },
          {
            "uid": "answer_uid_004",
            "answer_text": "no"
          }
        ]
      }
    ]
  }
}

### CODE REVIEW

**Comment by markphi2013** [comment]
@overnin : iam stuck on three main parts
1. what do you think about the naming of the function/view and controller?
2. i added the json view but i do remember/know how to turn the the different variable used in the Json call into usable values like program name, url and database.
screenshot from 2016-08-16 16 13 18
3. From the screen shot above when i run the post the program saves but i get error as shown above

File: app/Controller/ProgramRemoteSurveryController.php [hunk 1]
```diff
@@ -19,12 +19,82 @@ class ProgramRemoteSurveryController extends AppController
...
+    
+    protected function _instanciateVumiRabbitMQ()
+    {
+        $this->VumiRabbitMQ = new VumiRabbitMQ(Configure::read('vusion.rabbitmq'));
+    }
     
     
     public function addSurvery()
     {
+        $requestSuccess = false;
```
Comment (overnin) [inline app/Controller/ProgramRemoteSurveryController.php]: it seems that you are copy pasting lot of lines from the programController. I would urge you to rather create a Component class. In this component class you would have a few functions like "AddProgram()" or "_ajaxDataPatch"

File: app/Config/routes.php [hunk 2]
```diff
@@ -43,8 +43,8 @@
 	Router::mapResources('programRequests', array('prefix' => '/:program/'));
 	Router::mapResources('programParticipants', array('prefix'=> '/:program/'));
 	Router::mapResources('Programs');
-	Router::mapResources('ProgramRemoteSurvery');
-	Router::mapResources('ProgramsAjax2', array('prefix'=> '/:program/'));
+	Router::mapResources('ProgramsRemoteSurvery');
```
Comment (overnin) [inline app/Config/routes.php]: the naming of this controller doesn't seem to be consistent with rest of the controller. the prefix "ProgramXXX" is supposed to be used for controller that are used WITHIN an existing program, and this is expressed in the routing as the ProgramXXX controllers require to have the url of the program "array('prefix'=> '/:program/')"

I would rather suggest that you would name the controller **InstantSurvey**

**Comment by markphi2013** [comment]
Iam having a problem writing the test?
Also having the import of participants CSV on the call of this API?

**Comment by overnin** [comment]
For the test what is the problem?  - how to reproduce, screen shot, .... make it easy for me please -

import from where? what import what? is the url to download the participant list provided? give details please, i cannot guess

**Comment by markphi2013** [comment]
http://ask-people.vizzuality.com/api/docs/ 

Updated askpeople api doc @overnin

**Comment by markphi2013** [comment]
New updated API get call  for new survery

```
{
  "id": 2709,
  "participants_url": "/api/surveys/2709/reporters/csv",
  "questions": [
    {
      "id": 7209,
      "question_type": "select",
      "question_text": "Authentic beard yuccie vinegar.?",
      "answers": [
        {
          "id": 21442,
          "answer_text": "austin"
        },
        {
          "id": 21443,
          "answer_text": "chia"
        },
        {
          "id": 21444,
          "answer_text": "lumbersexual"
        }
      ]
    },
    {
      "id": 7210,
      "question_type": "select",
      "question_text": "Try-hard mixtape organic tousled yuccie iphone disrupt quinoa bitters.?",
      "answers": [
        {
          "id": 21445,
          "answer_text": "mumblecore"
        },
        {
          "id": 21446,
          "answer_text": "microdosing"
        },
        {
          "id": 21447,
          "answer_text": "pour-over"
        }
      ]
    },
    {
      "id": 7211,
      "question_type": "select",
      "question_text": "Freegan organic disrupt asymmetrical intelligentsia beard.?",
      "answers": [
        {
          "id": 21448,
          "answer_text": "cronut"
        },
        {
          "id": 21449,
          "answer_text": "hashtag"
        },
        {
          "id": 21450,
          "answer_text": "hoodie"
        }
      ]
    }
  ]
}
```

**Comment by markphi2013** [comment]
@overnin  hello i have add a dialogue create by the api call please huv a look at the test and feedback me if this will work 
https://github.com/texttochange/vusion-frontend/pull/285

**Pull request closed without merging**